dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/router/script/ScriptRouter.java

```diff
@@ -114,13 +113,24 @@ public URL getUrl() {
         }
     }
 
+    @Override
+    public Integer getPriority() {
+        return priority;
+    }
+
     @Override
     public int compareTo(Router o) {
-        if (o == null || o.getClass() != ScriptRouter.class) {
-            return 1;
+        if (o == null) {

```

Comment:
shouldn't we call super first, and do extra logic when super return 0 instead?

Reply:
This does not match the old logic.
The old logic is: If the priorities are equal, compare the rules.

If you call super first and the priority is equal, continue to compare the url. If the url returns not 0, it will return directly.

Old logic doesn't compare to url and url would not effect result.

Reply:
Simply,
Sometimes the url compare(super method) return 1. But the rule compare return -1.
If we can super first, it will return 1.
But in the old logic, it will return -1.